.github/workflows/provenance.yaml

```diff
@@ -60,6 +60,12 @@ jobs:
           ./oak_functions/loader/bin/oak_functions_loader \
           ./scripts/runner build-functions-server
 
+      - name: Validate the shape of the generated provenance files
+        uses: cardinalby/schema-validator-action@v1
+        with:
+          file: './provenance/**/*.json'
+          schema: 'https://raw.githubusercontent.com/project-oak/transparent-release/main/schema/amber-slsa-buildtype/v1.json'
```

Comment:
+1 currently this is a "floating" dependency, at the very least it needs to be pinned to a specific commit, but that would also prevent us from making incompatible changes, which we may want to do while things are in flux. Anyways, we can have it and see if it becomes annoying, as long as we are not too attached to it in case we need to temporarily disable it, or find a different solution down the line.

Reply:
True to the floating dep, linking to a commit seems nicer. Relatedly, we could can also decide to add a "$schema" key to our actual attestations, which links to our schema.

Add that point the statement can be validated by any JSON validator using just the statement itself

Reply:
Also RE linking to commits, worth thinking about how this relates to the version numbers that we also maintain for our schemas. Seems redundant, at that point could consider just using commit hashes for versions

Reply:
Also the following snowclone comes to mind "All I want is content addressable hyperlinks, all we get is links to GitHub commit hashes" 😅

[insert plug for ent here]

Reply:
> [insert plug for ent here]

:heavy_check_mark:  https://github.com/google/ent

Reply:
FWIW I added it to the ent-index: https://github.com/tiziano88/ent-index/issues/1#issuecomment-1036113572 (instructions here: https://github.com/google/ent#updating-the-index )